src/Admin/AdminInterface.php

```diff
@@ -215,108 +181,80 @@ public function getIdParameter();
      * @return string a string representation of the identifiers for this instance
      */
-    public function getNormalizedIdentifier($entity);
+    public function getNormalizedIdentifier($entity): ?string;
 
     /**
      * Shorthand method for templating.
      *
      * @param object $entity
-     *
-     * @return mixed
      */
-    public function id($entity);
+    public function id($entity): ?string;
 
-    public function setValidator(ValidatorInterface $validator);
+    public function setValidator(ValidatorInterface $validator): void;
 
-    /**
-     * @return ValidatorInterface
-     */
-    public function getValidator();
+    public function getValidator(): ?ValidatorInterface;
 
-    /**
-     * @return FieldDescriptionCollection|null
-     */
-    public function getShow();
+    public function getShow(): ?FieldDescriptionCollection;
 
-    public function setFormTheme(array $formTheme);
+    public function setFormTheme(array $formTheme): void;
 
     /**
      * @return string[]
      */
-    public function getFormTheme();
+    public function getFormTheme(): array;
 
-    public function setFilterTheme(array $filterTheme);
+    public function setFilterTheme(array $filterTheme): void;
 
     /**
      * @return string[]
      */
-    public function getFilterTheme();
+    public function getFilterTheme(): array;
 
-    public function addExtension(AdminExtensionInterface $extension);
+    public function addExtension(AdminExtensionInterface $extension): void;
 
     /**
      * Returns an array of extension related to the current Admin.
      *
      * @return AdminExtensionInterface[]
      */
-    public function getExtensions();
+    public function getExtensions(): array;
 
-    public function setRouteBuilder(RouteBuilderInterface $routeBuilder);
+    public function setRouteBuilder(RouteBuilderInterface $routeBuilder): void;
 
-    /**
-     * @return RouteBuilderInterface
-     */
-    public function getRouteBuilder();
+    public function getRouteBuilder(): ?RouteBuilderInterface;
 
     /**
      * @param object $object
-     *
-     * @return string
      */
-    public function toString($object);
+    public function toString($object): string;
 
-    public function setLabelTranslatorStrategy(LabelTranslatorStrategyInterface $labelTranslatorStrategy);
+    public function setLabelTranslatorStrategy(LabelTranslatorStrategyInterface $labelTranslatorStrategy): void;
 
-    /**
-     * @return LabelTranslatorStrategyInterface
-     */
-    public function getLabelTranslatorStrategy();
+    public function getLabelTranslatorStrategy(): ?LabelTranslatorStrategyInterface;
 
     /**
      * Returning true will enable preview mode for
      * the target entity and show a preview button
      * when editing/creating an entity.
-     *
-     * @return bool
      */
-    public function supportsPreviewMode();
+    public function supportsPreviewMode(): bool;
 
     /**
      * @return object a new object instance
```

Comment:
I have in mind there is a PR currently open regarding this method. I'd prefer to have a clear base there before making changes here.

Reply:
Since you added the return type for the `BaseAdmin` class, I think it would be better to add a type here too.
In the PR opened, I gave solution to avoid using a null return value for getNewInstance so I don't think it will be merged. But if it was, they'll update the type when merging in master.

If we don't add return type hint, I'm afraid we forget to add it later.